packages/grafana-data/src/types/panel.ts

```diff
@@ -87,6 +109,9 @@ export interface PanelProps<T = any> {
   /** Panel options set by the user in the panel editor. Includes both default and custom panel options */
   options: T;
 
+  /**  * When set, it disregards the <options.tooltip> in favor of a fully customizable one.*/
+  getCustomTooltip?: (props: PanelCustomTimeSeriesTooltip) => React.ReactNode;
```

Comment:
here we have a problem with adding a type unique to TimeSeries visualization to the shared PanelProps interface that is used by all panel types. This is why this should be on options type that is unique to each visualization type.